nixos/modules/services/misc/metabase.nix

```diff
@@ -0,0 +1,117 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.services.metabase;
+
+  inherit (lib) mkEnableOption mkIf mkOption;
+  inherit (lib) optional optionalAttrs optionalString types;
+
+  dataDir = "/var/lib/metabase";
+
+in {
+
+  options = {
+
+    services.metabase = {
+      enable = mkEnableOption "Metabase service";
+
+      listen = {
+        ip = mkOption {
+          type = types.str;
+          default = "0.0.0.0";
+          description = ''
+            IP address that Metabase should listen on.
+          '';
+        };
+
+        port = mkOption {
+          type = types.port;
+          default = 3000;
+          description = ''
+            Listen port for Metabase.
+					'';
+        };
+      };
+
+			ssl = {
+				enable = mkOption {
+					type = types.bool;
+					default = false;
+					description = ''
+            Whether to enable SSL (https) support.
+					'';
+				};
+
+				port = mkOption {
+					type = types.port;
+					default = 8443;
+					description = ''
+            Listen port over SSL (https) for Metabase.
+					'';
+				};
```

Comment:
The formatting is messed up in this file, probably mixed up tabs and spaces (should use spaces in nixpkgs, and if you have an [editorconfig](https://editorconfig.org/) plugin this will be done automatically)